File: IGECAdmin/app/src/main/java/com/example/igec_admin/Activities/MainActivity.java
package com.example.igec_admin.Activities;


import android.Manifest;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.os.Environment;
import android.provider.Settings;
import android.view.MenuItem;
import android.view.MotionEvent;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.appcompat.widget.Toolbar;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.viewpager.widget.ViewPager;
import androidx.viewpager2.widget.ViewPager2;
import androidx.work.OneTimeWorkRequest;
import androidx.work.WorkRequest;

import com.example.igec_admin.Adatpers.ViewPagerAdapter;
import com.example.igec_admin.Dialogs.ProjectFragmentDialog;
import com.example.igec_admin.Fragments.AddMachineFragment;
import com.example.igec_admin.Fragments.AddProjectFragment;
import com.example.igec_admin.Fragments.AddUserFragment;
import com.example.igec_admin.Fragments.MachinesFragment;
import com.example.igec_admin.Fragments.ProjectsFragment;
import com.example.igec_admin.Fragments.UsersFragment;
import com.example.igec_admin.Fragments.VacationRequestsFragment;
import com.example.igec_admin.Fragments.VacationsLogFragment;
import com.example.igec_admin.R;
import com.example.igec_admin.Fragments.SummaryFragment;
import com.google.android.material.navigation.NavigationView;

import java.util.List;

import pub.devrel.easypermissions.AfterPermissionGranted;
import pub.devrel.easypermissions.AppSettingsDialog;
import pub.devrel.easypermissions.EasyPermissions;

public class MainActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {

    private DrawerLayout vDrawerLayout;
    private NavigationView vNavigationView;
    private ViewPager viewPager;
    private Toolbar vToolbar;

    // Vars
    private ActionBarDrawerToggle actionBarDrawerToggle;
    private int selectedTab = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        getExternalStoragePerm();
        super.onCreate(savedInstanceState);
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
        setContentView(R.layout.activity_main);
        isNetworkAvailable();
        initialize();

        // Listeners
        vDrawerLayout.addDrawerListener(actionBarDrawerToggle);
        vDrawerLayout.addDrawerListener(drawerListener);
        viewPager.addOnPageChangeListener(viewPagerListener);


        NavigationView navigationView = findViewById(R.id.navView);


        TextView contact_info = navigationView.getHeaderView(0).findViewById(R.id.contact_info);
        contact_info.setOnClickListener(view -> {
            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://" + getString(R.string.nav_header_subtitle)));
            this.startActivity(intent);
        });

    }

    @Override
    protected void onStop() {
        super.onStop();
        AddProjectFragment.clearTeam();
        ProjectFragmentDialog.clearTeam();
    }

    @Override
    protected void onResume() {
        super.onResume();
        isNetworkAvailable();
    }

    //Functions

    private void getExternalStoragePerm() {
        if (Build.VERSION.SDK_INT >= 30) {
            if (!Environment.isExternalStorageManager()) {
                Intent intent = new Intent();
                intent.setAction(Settings.ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION);
                Uri uri = Uri.fromParts("package", this.getPackageName(), null);
                intent.setData(uri);
                startActivity(intent);
            }
        }
    }

    private void isNetworkAvailable() {
        ConnectivityManager connectivityManager
                = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        boolean connected = activeNetworkInfo != null && activeNetworkInfo.isConnected();
        if (!connected) {
            Intent intent = new Intent(MainActivity.this, SplashScreen_InternetConnection.class);
            startActivity(intent);
            finish();
        }
    }

    private void initialize() {
        // Views
        vToolbar = findViewById(R.id.toolbar);
        vDrawerLayout = findViewById(R.id.drawer);
        vNavigationView = findViewById(R.id.navView);
        viewPager = findViewById(R.id.fragment_container);

        setSupportActionBar(vToolbar);
        actionBarDrawerToggle = new ActionBarDrawerToggle(this, vDrawerLayout, vToolbar, R.string.openNavBar, R.string.closeNavBar);
        actionBarDrawerToggle.syncState();
        vNavigationView.setNavigationItemSelectedListener(this);


        AddUserFragment addUserFragment = new AddUserFragment();
        AddProjectFragment addProjectFragment = new AddProjectFragment();
        AddMachineFragment addMachineFragment = new AddMachineFragment();
        UsersFragment usersFragment = new UsersFragment();
        ProjectsFragment projectsFragment = new ProjectsFragment();
        MachinesFragment machinesFragment = new MachinesFragment();
        SummaryFragment summaryFragment = new SummaryFragment();
        VacationRequestsFragment vacationRequestsFragment = new VacationRequestsFragment();
        VacationsLogFragment vacationsLogFragment = new VacationsLogFragment();

        ViewPagerAdapter viewPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager(), 0);
        viewPagerAdapter.addFragment(addUserFragment, getString(R.string.add_user));
        viewPagerAdapter.addFragment(addProjectFragment, getString(R.string.add_project));
        viewPagerAdapter.addFragment(addMachineFragment, getString(R.string.add_machine));
        viewPagerAdapter.addFragment(usersFragment, getString(R.string.users));
        viewPagerAdapter.addFragment(projectsFragment, getString(R.string.projects));
        viewPagerAdapter.addFragment(machinesFragment, getString(R.string.machines));
        viewPagerAdapter.addFragment(vacationRequestsFragment, getString(R.string.vacation_requests));
        viewPagerAdapter.addFragment(vacationsLogFragment, getString(R.string.vacations_log));
        viewPagerAdapter.addFragment(summaryFragment, getString(R.string.summary));

        viewPager.setAdapter(viewPagerAdapter);
        vNavigationView.getMenu().getItem(0).setChecked(true);


    }

    @Override
    public void onBackPressed() {
        if (vDrawerLayout.isDrawerOpen(GravityCompat.START))
            vDrawerLayout.closeDrawer(GravityCompat.START);
        else
            finish();
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        vDrawerLayout.closeDrawer(GravityCompat.START);
        selectedTab = item.getItemId();
        return true;
    }

    private void changeTab(int itemId) {
        if (itemId == R.id.item_addUser)
            viewPager.setCurrentItem(0, true);
        if (itemId == R.id.item_addProject)
            viewPager.setCurrentItem(1, true);
        if (itemId == R.id.item_addMachine)
            viewPager.setCurrentItem(2, true);
        if (itemId == R.id.item_Users)
            viewPager.setCurrentItem(3, true);
        if (itemId == R.id.item_Projects)
            viewPager.setCurrentItem(4, true);
        if (itemId == R.id.item_Machines)
            viewPager.setCurrentItem(5, true);
        if (itemId == R.id.item_Vacation_Requests)
            viewPager.setCurrentItem(6, true);
        if (itemId == R.id.item_VacationsLog)
            viewPager.setCurrentItem(7, true);
        if (itemId == R.id.item_summary)
            viewPager.setCurrentItem(8, true);
    }

    private ViewPager.OnPageChangeListener viewPagerListener = new ViewPager.OnPageChangeListener() {
        @Override
        public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

        }

        @Override
        public void onPageSelected(int position) {
            vNavigationView.getMenu().getItem(position).setChecked(true);
            vToolbar.setTitle(vNavigationView.getMenu().getItem(position).getTitle());
        }

        @Override
        public void onPageScrollStateChanged(int state) {

        }
    };
    private DrawerLayout.DrawerListener drawerListener = new DrawerLayout.DrawerListener() {
        @Override
        public void onDrawerSlide(@NonNull View drawerView, float slideOffset) {

        }

        @Override
        public void onDrawerOpened(@NonNull View drawerView) {

        }

        @Override
        public void onDrawerClosed(@NonNull View drawerView) {
            changeTab(selectedTab);
        }

        @Override
        public void onDrawerStateChanged(int newState) {

        }
    };


}

File: IGECAdmin/admin/src/main/java/com/igec/admin/Activities/MainActivity.kt
package com.igec.admin.Activities

import android.content.Intent
import android.net.ConnectivityManager
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.os.Environment
import android.provider.Settings
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.igec.admin.databinding.ActivityMainBinding
import com.igec.admin.Dialogs.ProjectFragmentDialog
import com.igec.admin.Fragments.*

import com.google.android.material.navigation.NavigationView
import com.igec.admin.R

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var binding: ActivityMainBinding
    private lateinit var navController: NavController
    private var lastTab: Int = R.id.nav_add_user
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.toolbar)

        val drawerLayout: DrawerLayout = binding.drawerLayout
        val navView: NavigationView = binding.navView
        val navHostFragment =
            supportFragmentManager.findFragmentById(R.id.nav_host_fragment_content_main) as NavHostFragment
        navController = navHostFragment.navController
        // Passing each menu ID as a set of Ids because each
        // menu should be considered as top level destinations.
        appBarConfiguration = AppBarConfiguration(
            setOf(
                R.id.nav_add_user, R.id.nav_add_project, R.id.nav_add_machine,
                R.id.nav_users, R.id.nav_projects, R.id.nav_machines,
                R.id.nav_vacation_requests, R.id.nav_vacations_log, R.id.nav_summary,
            ), drawerLayout
        )
        setupActionBarWithNavController(navController, appBarConfiguration)
        navView.setupWithNavController(navController)
        navView.setNavigationItemSelectedListener(this)


        val navigationView = findViewById<NavigationView>(R.id.nav_view)


        val contact_info = navigationView.getHeaderView(0).findViewById<TextView>(R.id.contact_info)
        contact_info.setOnClickListener { view: View? ->
            val intent = Intent(
                Intent.ACTION_VIEW,
                Uri.parse("https://" + getString(R.string.nav_header_subtitle))
            )
            this@MainActivity.startActivity(intent)
        }
    }

    override fun onStop() {
        super.onStop()
        AddProjectFragment.clearTeam()
        ProjectFragmentDialog.clearTeam()
    }
    override fun onResume() {
        super.onResume()
        isNetworkAvailable()
    }

    //Functions
    private fun getExternalStoragePerm() {
        if (Build.VERSION.SDK_INT >= 30) {
            if (!Environment.isExternalStorageManager()) {
                val intent = Intent()
                intent.action = Settings.ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION
                val uri = Uri.fromParts("package", this.packageName, null)
                intent.data = uri
                startActivity(intent)
            }
        }
    }

    private fun isNetworkAvailable() {
        val connectivityManager = getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetworkInfo = connectivityManager.activeNetworkInfo
        val connected = activeNetworkInfo != null && activeNetworkInfo.isConnected
        if (!connected) {
            val intent = Intent(this@MainActivity, SplashScreen_InternetConnection::class.java)
            startActivity(intent)
            finish()
        }
    }
//    override fun onCreateOptionsMenu(menu: Menu): Boolean {
//        // Inflate the menu; this adds items to the action bar if it is present.
//        menuInflater.inflate(R.menu.main_activity2, menu)
//        return true
//    }

    override fun onSupportNavigateUp(): Boolean {
        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        binding.drawerLayout.addDrawerListener(object : DrawerLayout.DrawerListener {
            override fun onDrawerSlide(drawerView: View, slideOffset: Float) {}
            override fun onDrawerOpened(drawerView: View) {}
            override fun onDrawerStateChanged(newState: Int) {}
            override fun onDrawerClosed(drawerView: View) {
                // This method will be called after drawer animation finishes
                // Perform the fragment replacement
                if(lastTab != item.itemId) {
                    if (lastTab == R.id.nav_add_project) {
                        AddProjectFragment.clearTeam()
                    }
                    when (item.itemId) {
                        R.id.nav_add_user -> {
                            binding.toolbar.title = getString(R.string.add_user)
                            supportFragmentManager.beginTransaction()
                                .replace(R.id.nav_host_fragment_content_main, AddUserFragment())
                                .addToBackStack(null)
                                .commit()
                        }
                        R.id.nav_add_project -> {
                            binding.toolbar.title = getString(R.string.add_project)
                            supportFragmentManager.beginTransaction()
                                .replace(R.id.nav_host_fragment_content_main, AddProjectFragment())
                                .addToBackStack(null)
                                .commit()
                        }
                        R.id.nav_add_machine -> {
                            binding.toolbar.title = getString(R.string.add_machine)
                            supportFragmentManager.beginTransaction()
                                .replace(R.id.nav_host_fragment_content_main, AddMachineFragment())
                                .addToBackStack(null)
                                .commit()
                        }

                        R.id.nav_users -> {
                            binding.toolbar.title = getString(R.string.users)
                            supportFragmentManager.beginTransaction()
                                .replace(R.id.nav_host_fragment_content_main, UsersFragment())
                                .addToBackStack(null)
                                .commit()
                        }
                        R.id.nav_projects -> {
                            binding.toolbar.title = getString(R.string.projects)
                            supportFragmentManager.beginTransaction()
                                .replace(R.id.nav_host_fragment_content_main, ProjectsFragment())
                                .addToBackStack(null)
                                .commit()
                        }
                        R.id.nav_machines -> {
                            binding.toolbar.title = getString(R.string.machines)
                            supportFragmentManager.beginTransaction()
                                .replace(R.id.nav_host_fragment_content_main, MachinesFragment())
                                .addToBackStack(null)
                                .commit()
                        }

                        R.id.nav_vacation_requests -> {
                            binding.toolbar.title = getString(R.string.vacation_requests)
                            supportFragmentManager.beginTransaction()
                                .replace(
                                    R.id.nav_host_fragment_content_main,
                                    VacationRequestsFragment()
                                )
                                .addToBackStack(null)
                                .commit()
                        }
                        R.id.nav_vacations_log -> {
                            binding.toolbar.title = getString(R.string.vacations_log)
                            supportFragmentManager.beginTransaction()
                                .replace(
                                    R.id.nav_host_fragment_content_main,
                                    VacationsLogFragment()
                                )
                                .addToBackStack(null)
                                .commit()
                        }
                        R.id.nav_summary -> {
                            binding.toolbar.title = getString(R.string.summary)
                            supportFragmentManager.beginTransaction()
                                .replace(R.id.nav_host_fragment_content_main, SummaryFragment())
                                .addToBackStack(null)
                                .commit()
                        }
                    }
                    lastTab = item.itemId
                }
                // Cross fade back the content container and hide progress bar


                // Remove this listener so close by, for example, swiping do not call it again
                binding.drawerLayout.removeDrawerListener(this)
            }
        })

        // Closes the drawer, triggering the listener above
        binding.drawerLayout.closeDrawer(GravityCompat.START)
        return true
    }

    override fun onBackPressed() {
        if (binding.drawerLayout.isDrawerOpen(GravityCompat.START))
            binding.drawerLayout.closeDrawer(GravityCompat.START)
        else
            finish()
    }
}